Implement equality operation for shares

Drake-Eidukas opened the pull request:
I discussed with Hanyun, and she's super bogged down with thesis work, so I implemented some of the refactors suggested from @amiller  in #237

amiller commented on honeybadgermpc/equality.py:
```diff
@@ -0,0 +1,64 @@
+        x_first, x_left = context.ShareArray(x[::2]), context.ShareArray(x[1::2])
+        x = (await (x_first * x_left))._shares
+
+    return await x[0].open() != 0
```
just `return x[0]` here. Equality should return a secret share of `[a == b]`, it's up to the caller if they wish to publicly reconstruct it

Drake-Eidukas replied:
Sounds good.

Drake-Eidukas commented:
@amiller  should be good to go now

Drake-Eidukas committed:
Implements equality operation for shares

This PR also adds in a flag to travis to run the tests sequentially
for each test file to prevent weird hanging conditions we were getting
when running on travis.

Drake-Eidukas commented:
This resolves one of the three points in #27
